### ISSUE TITLE
CPU usage spikes past 100%, even with torch.set_num_threads(1)

### ISSUE THREAD

**Issue by bakermanbrian** [pytorch/pytorch#102252]
### 🐛 Describe the bug

I'm running a workflow with two external libraries that utilize torch (pyannote & whisper). When I try to run it, the CPU usage spikes past 100%, even when torch.set_num_threads(1) is set. Below is the minimum reproducible sample. The issue is occurring on a GCP project. Full workflow reproduction here:
https://github.com/bakermanbrian/gcpwhisppyan

Minimum example:
```
import whisper
import os
from stable_whisper import modify_model
from pyannote.audio import Pipeline
import time
import torch

torch.set_num_threads(1)
model = whisper.load_model("base")
modify_model(model)


diarizationModel = Pipeline.from_pretrained("pyannote/speaker-diarization",
	                                    use_auth_token="<AUTH TOKEN>")
diarizationModel=diarizationModel.to(0)

def transcription(filePath):
	modelRes=model.transcribe(filePath).to_dict()
	res = {"text": modelRes["text"], "language": modelRes["language"]}
	toAdd=[]
	diarizationResult=diarizationModel(filePath)
	for turn, _, speaker in diarizationResult.itertracks(yield_label=True):
		toAdd.append({"startTime": turn.start, "stopTime": turn.end, "speaker": speaker})
	res["diarization"]=toAdd
        return res
    
print(transcription(""))
```

Requirements.txt
```
asteroid-filterbanks >=0.4
einops >=0.6.0
huggingface_hub >= 0.13.0
lightning >= 2.0.1
omegaconf >=2.1,<3.0
pyannote.core >= 5.0.0
pyannote.database >= 5.0.1
pyannote.metrics >= 3.2
pyannote.pipeline >= 2.3   # 2.4
pytorch_metric_learning >= 2.1.0
rich >= 12.0.0
semver >= 3.0.0
soundfile >= 0.12.1
speechbrain >= 0.5.14
tensorboardX >= 2.6
torch >= 2.0.0
torch_audiomentations >= 0.11.0
torchaudio >= 2.0.0
torchmetrics >= 0.11.0

#replaced torch with one above
numba
numpy
tqdm
more-itertools
tiktoken==0.3.3


fastapi
uvicorn
librosa
python-multipart
audioread
gunicorn
```

Docker file for the env
```
FROM nvidia/cuda:12.0.0-runtime-ubuntu22.04

EXPOSE 8000
RUN apt-get update && apt-get install -y --no-install-recommends \
    python3.9 \
    python3-pip \
    && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*

WORKDIR /
COPY . .
RUN apt-get update
RUN apt-get install --assume-yes git
RUN apt update && apt install -y ffmpeg
RUN pip install --upgrade pip
RUN pip install git+https://github.com/openai/whisper.git

RUN pip install git+https://github.com/jianfch/stable-ts.git

RUN pip install -qq https://github.com/pyannote/pyannote-audio/archive/refs/heads/develop.zip

RUN pip install -r requirements.txt

```

### Versions

-

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

**Comment by mingfeima** [comment]
@jingxu10 could you please help debug this one ?
@bakermanbrian i can not access this link https://github.com/bakermanbrian/gcpwhisppyan, could you please verify ? Also is there any multi-threading libs used in the workload ? for example, [multiprocessing] ?(https://pytorch.org/docs/stable/notes/multiprocessing.html)

**Comment by bakermanbrian** [comment]
Appreciate you looking into this, but I resolved this :) 

Turns out the fix was to use pytorch/pytorch:2.0.1-cuda11.7-cudnn8-runtime as my base build.
